python/cudf/cudf/core/join/join.py

```diff
@@ -0,0 +1,431 @@
+        self.typecast_input_to_libcudf()
+        libcudf_result = libcudf.join.join(
+            self.lhs,
+            self.rhs,
+            self.left_on,
+            self.right_on,
+            self.how,
+            self.method,
+            left_index=self.left_index,
+            right_index=self.right_index,
+        )
+        result = self.typecast_libcudf_to_output(libcudf_result, output_dtypes)
+        return result
+
+    def preprocess_merge_params(self, on, left_on, right_on, lsuffix, rsuffix):
+        if on:
+            on = [on] if isinstance(on, str) else list(on)
+            left_on = right_on = on
+        else:
+            on = []
+        if left_on:
+            left_on = [left_on] if isinstance(left_on, str) else list(left_on)
+        else:
+            left_on = []
+        if right_on:
+            right_on = (
+                [right_on] if isinstance(right_on, str) else list(right_on)
+            )
+        else:
+            right_on = []
+
+        same_named_columns = set(self.lhs._data.keys()) & set(
+            self.rhs._data.keys()
+        )
+        if not (left_on or right_on) and not (
+            self.left_index and self.right_index
+        ):
+            left_on = right_on = list(same_named_columns)
+
+        no_suffix_cols = []
+        for name in same_named_columns:
+            if left_on is not None and right_on is not None:
+                if name in left_on and name in right_on:
+                    if left_on.index(name) == right_on.index(name):
+                        no_suffix_cols.append(name)
+
+        for name in same_named_columns:
+            if name not in no_suffix_cols:
+                self.lhs.rename({name: "%s%s" % (name, lsuffix)}, inplace=True)
+                self.rhs.rename({name: "%s%s" % (name, rsuffix)}, inplace=True)
+                if name in left_on:
+                    left_on[left_on.index(name)] = "%s%s" % (name, lsuffix)
+                if name in right_on:
+                    right_on[right_on.index(name)] = "%s%s" % (name, rsuffix)
+
+        self.left_on = left_on if left_on is not None else []
+        self.right_on = right_on if left_on is not None else []
+        self.lsuffix = lsuffix
+        self.rsuffix = rsuffix
+
+    @staticmethod
+    def validate_merge_cfg(
+        lhs,
+        rhs,
+        on,
+        left_on,
+        right_on,
+        left_index,
+        right_index,
+        how,
+        lsuffix,
+        rsuffix,
+    ):
+        """
+        Error for various combinations of merge input parameters
+        """
+
+        # must actually support the requested merge type
+        if how not in ["left", "inner", "outer", "leftanti", "leftsemi"]:
+            raise NotImplementedError(
+                "{!r} merge not supported yet".format(how)
+            )
+
+        # Passing 'on' with 'left_on' or 'right_on' is ambiguous
+        if on is not None:
+            if left_on is not None or right_on is not None:
```

Comment:
Combine these two:
`if on is not None and (left_on is not None or right_on is not None):`

Reply:
Done